rasa/core/interpreter.py

```diff
@@ -296,6 +301,146 @@ def _load_interpreter(self) -> None:
 
         self.interpreter = Interpreter.load(self.model_directory)
 
+MODEL_NAME_PREFIX = "nlu"
+
+def find_same(action_name, attribute, training_data):
+        for ex in training_data.training_examples:
+            if ex.as_dict()['text'] == action_name and ex.get(SPARSE_FEATURE_NAMES[attribute]) is not None:
+                example = ex
+                break
+        return example
+
+class RasaCoreInterpreter(NaturalLanguageInterpreter):
+    def __init__(self,):
+        self._load_interpreter()
+
+    def parse(self, text: Text):
+        result = self.interpreter.parse(text, only_output_properties=False)
+        return result
+
+    def _load_interpreter(self) -> None:
+        from rasa.nlu import config
+
+        config = config.load()
+        self.trainer = Trainer(config)
+        self.interpreter = Interpreter(self.trainer.pipeline, {})
+
+    def prepare_training_data(self, trackers_as_states, trackers_as_actions, domain):
+        training_data = []
+        for tracker in trackers_as_states:
+            for state in tracker: 
+                if state and not state == {}:
+                    training_data.append(state["prev_action"])
+                    if "user" in state.keys():
+                        training_data.append(state["user"])
+        training_data += [Message(action) for action in domain.action_names]
+        tr_data = TrainingData(training_examples=training_data)
```

Comment:
let's use appropriate names:
```suggestion
        training_examples += [Message(action) for action in domain.action_names]
        training_data = TrainingData(training_examples)
```

Reply:
above and below as well